### ISSUE TITLE
Bug: Mathjax v3 - Certain Symbols (€) throw a TexParser error

### ISSUE THREAD

**Issue by adamc-texthelp** [mathjax/MathJax#3033]
Using certain symbols causes Mathjax v3 to throw an error in the TexParser. I'm sure there's more examples, but I've noticed it with a couple of currency symbols such as:

€ - euro
₣ - Francs
₹ - Rupee

You can get around the error by wrapping the symbols in a `\text{}` environment, but this wasn't necessary in mathjax v2 and I'm unsure what symbols might cause this issue.

### Steps to Reproduce
You can create this error on the tex2svg demo sample provided in this repo.
https://mathjax.github.io/MathJax-demos-web/input-tex2svg.html

Can also be seen on the tex2chtml demo
https://mathjax.github.io/MathJax-demos-web/input-tex2chtml.html

Alternatively I recreated the tex2svg in jsfiddle, but I log the error to the console.
https://jsfiddle.net/hLbp1z4s/

1. Use one of the symbols mentioned above (e.g. €12) and press "Render TeX"
2. An error is thrown `Cannot read properties of null (reading '4')`

### Technical details
Example pages use latest Mathjax@3, I've also confirmed the issue with `mathjax-full@3.2.2`

I've seen the issue in both Google Chrome, Edge and Firefox 

### Questions

Any recommendations on a workaround for this in the meantime?

**Comment by dpvc** [comment]
This is fixed in the next release of v4 (via mathjax/MathJax-src#900).  For now, you can use

``` js
  MathJax = {
    tex: {
      packages: {'[+]': ['fix-unicode']}
    },
    startup: {
      ready() {
        const {Configuration} = MathJax._.input.tex.Configuration;
        const {MapHandler} = MathJax._.input.tex.MapHandler;
        const NodeUtil = MathJax._.input.tex.NodeUtil.default;
        const {getRange} = MathJax._.core.MmlTree.OperatorDictionary;
        function Other(parser, char) {
          const font = parser.stack.env['font'];
          let def = font ? {mathvariant: parser.stack.env['font']} : {};
          const remap = (MapHandler.getMap('remap')).lookup(char);
          const range = getRange(char);
          const type = range?.[3] || 'mo';
          let mo = parser.create('token', type, def, (remap ? remap.char : char));
          range?.[4] && mo.attributes.set('mathvariant', range[4]);
          if (type === 'mo') {
            NodeUtil.setProperty(mo, 'fixStretchy', true);
            parser.configuration.addNode('fixStretchy', mo);
          }
          parser.Push(mo);
        }
        Configuration.create('fix-unicode', {fallback: {character: Other}});
        MathJax.startup.defaultReady();
      }
    }
  };
```

to apply the needed fix.

**Comment by dpvc** [comment]
PS, the error was using `range[4]` rather than `range?.[4]` in the middle of the `Other()` function.

**Comment by adamc-texthelp** [comment]
Appreciate the quick and helpful response. We will definitely move up to Mathjax 4 when it's released. Would you be able to provide an example of that workaround for `mathjax-full`?

I created a cutdown sample of how we use `mathjax-full`, but I'm unsure how to add that `startup` logic to replace the broken  `range[4]` check.

https://codesandbox.io/s/distracted-roman-23zpum?file=/src/index.js

**Comment by dpvc** [comment]
Here is a modified version of your code:

``` js
import { mathjax } from "mathjax-full/js/mathjax";
import { TeX } from "mathjax-full/js/input/tex";
import { AllPackages } from "mathjax-full/js/input/tex/AllPackages";
import { SVG } from "mathjax-full/js/output/svg";
import { RegisterHTMLHandler } from "mathjax-full/js/handlers/html";
import { chooseAdaptor } from "mathjax-full/js/adaptors/chooseAdaptor";

import {Configuration} from "mathjax-full/js/input/tex/Configuration";
import {MapHandler} from "mathjax-full/js/input/tex/MapHandler";
import NodeUtil from "mathjax-full/js/input/tex/NodeUtil";
import {getRange} from "mathjax-full/js/core/MmlTree/OperatorDictionary";

const adaptor = chooseAdaptor();
RegisterHTMLHandler(adaptor);

function Other(parser, char) {
  const font = parser.stack.env['font'];
  let def = font ? {mathvariant: parser.stack.env['font']} : {};
  const remap = (MapHandler.getMap('remap')).lookup(char);
  const range = getRange(char);
  const type = range?.[3] || 'mo';
  let mo = parser.create('token', type, def, (remap ? remap.char : char));
  range?.[4] && mo.attributes.set('mathvariant', range[4]);
  if (type === 'mo') {
    NodeUtil.setProperty(mo, 'fixStretchy', true);
    parser.configuration.addNode('fixStretchy', mo);
  }
  parser.Push(mo);
}
Configuration.create('fix-unicode', {fallback: {character: Other}});

const tex = new TeX({
  formatError(jax, error) {
    throw Error(`TeX error: ${error.message}`);
  },
  packages: AllPackages.filter((x) => x !== "textmacros").concat('fix-unicode')
});

const svgOutput = new SVG({ fontCache: "none" });

// https://github.com/mathjax/MathJax/issues/2824#issuecomment-1020447016
// use document for first parameter so mathjax can get the font size correct for non TeX fonts
const html = mathjax.document(document, {
  InputJax: tex,
  OutputJax: svgOutput
});

export const toSVG = (latex) => {
  return html.convert(latex, { display: true });
};

const svgHtml = toSVG("12345€");
const wrapper = document.createElement("div");
wrapper.append(svgHtml);
document.body.append(wrapper);
```

Note that you don't have to subclass the TeX input jax in order to get the error handling, as there is a `formatError()` option that you can use, as I have done here.  Also, since you are adding the SVG into the page, there is no need to convert to `outerHTML` and then `innerHTML` again, and that has adverse consequences (like losing the event handlers).  But it may be that this test page does not reflect your actual usage of the resulting SVG.

Finally, the `chooseAdaptor()` is really only for development use, where you want to be able to run the exact same code in both the browser and in node.  If you use it in production, you will package both the HTML adaptor and the LiteDOM adaptor, and so will be shipping extra code that you won't be using.

**Comment by dpvc** [comment]
Fixed in v4.0.
